File: logger/src/main/java/ru/digipeople/logger/LogcatLoggerFactory.java
package ru.digipeople.logger;

/**
 * Default Logcat implementation of {@link ILoggerFactory}.
 *
 * @author Aleksandr Brazhkin
 */
public class LogcatLoggerFactory implements ILoggerFactory {
    @Override
    public Logger getLogger(String name) {
        return new LogcatLogger(name);
    }

    @Override
    public Logger getLogger(Class<?> clazz) {
        return new LogcatLogger(clazz.getSimpleName());
    }
}

File: logger_kotlin/src/main/java/ru/digipeople/logger/LogcatLoggerFactory.kt
package ru.digipeople.logger

import kotlin.reflect.KClass

/**
 * Default Logcat implementation of [LoggerFactory].
 *
 * @author Aleksandr Brazhkin
 */
class LogcatLoggerFactory : ILoggerFactory {
    override fun getLogger(name: String): Logger {
        return LogcatLogger(name)
    }

    override fun getLogger(clazz: Class<*>): Logger {
        return LogcatLogger(clazz.simpleName)
    }

    override fun getLogger(kClass: KClass<*>): Logger {
        return LogcatLogger(kClass.java.simpleName)
    }
}
